bika/lims/browser/sample/view.py

```diff
@@ -514,3 +522,176 @@ def _schedule_sampling_permissions(self):
         roles = member.getRoles()
         return self.context.bika_setup.getScheduleSamplingEnabled() and\
             ('SamplingCoordinator' in roles or 'Manager' in roles)
+
+    def isItemAllowed(self, obj):
+        """
+        Checks the BikaLIMS conditions and also checks filter bar conditions
+        @Obj: it is a sample object.
+        @return: boolean
+        """
+        # TODO-performance:we are expecting for the sample object.
+        # Should be only a brain
+        if self.filter_bar_enabled and not self.filter_bar_check_item(obj):
+            return False
+        return super(SamplesView, self).isItemAllowed(obj)
+
+    def getFilterBar(self):
+        """
+        This function creates an instance of BikaListingFilterBar if the
+        class has not created one yet.
+        :return: a BikaListingFilterBar instance
+        """
+        self._advfilterbar = self._advfilterbar if self._advfilterbar else \
+            BikaListingFilterBar(context=self.context, request=self.request)
+        return self._advfilterbar
+
+
+class BikaListingFilterBar(BaseBikaListingFilterBar):
+    """
+    This class defines a filter bar to make advanced queries in
+    BikaListingView. This filter shouldn't override the 'filter by state'
+    functionality
+    """
+    # TODO-performance: Improve filter bar using catalogs
+    def filter_bar_builder(self):
+        """
+        The template is going to call this method to create the filter bar in
+        bika_listing_filter_bar.pt
+        If the method returns None, the filter bar will not be shown.
+        :return: a list of dictionaries as the filtering fields or None.
+        """
+        fields_dict = [{
+            'name': 'sample_condition',
+            'label': _('Sample condition'),
+            'type': 'select',
+            'voc': self._getSampleConditionsVoc(),
+        }, {
+            'name': 'print_state',
+            'label': _('Print state'),
+            'type': 'select',
+            'voc': self._getPrintStatesVoc(),
+        }, {
+            'name': 'sample_type',
+            'label': _('Sample type'),
+            'type': 'select',
+            'voc': self._getSampleTypesVoc(),
+        }, {
+            'name': 'case',
+            'label': _('Cases'),
+            'type': 'autocomplete_text',
+            'voc': json.dumps(self._getCasesVoc()),
+        }, {
+            'name': 'date_received',
+            'label': _('Date received'),
+            'type': 'date_range',
+        },
+        ]
+        return fields_dict
```

Comment:
This is very similar to the filterbar for analysisrequests. Maybe we could generate a superclass (say "SampleARBikaListingFilterBar" and move there all these repetitive functions. Then you'll only need to create an inherited class with minimum code in both AnalysisRequestsView and SampleView